Xamarin.Essentials/Launcher/Launcher.mac.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using AppKit;
+using Foundation;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Launcher
+    {
+        static Task<bool> PlatformCanOpenAsync(Uri uri) =>
+            Task.FromResult(NSWorkspace.SharedWorkspace.UrlForApplication(GetNativeUrl(uri)) != null);
+
+        static Task PlatformOpenAsync(Uri uri) =>
+            Task.FromResult(NSWorkspace.SharedWorkspace.OpenUrl(GetNativeUrl(uri)));
+
+        static Task<bool> PlatformTryOpenAsync(Uri uri)
+        {
+            var nativeUrl = GetNativeUrl(uri);
+            var canOpen = NSWorkspace.SharedWorkspace.UrlForApplication(nativeUrl) != null;
+
+            if (canOpen)
+            {
+                // TODO: there is an OpenUrlAsync that may be useful
+                return Task.FromResult(NSWorkspace.SharedWorkspace.OpenUrl(nativeUrl));
+            }
+
+            return Task.FromResult(canOpen);
+        }
+
+        internal static NSUrl GetNativeUrl(Uri uri)
```

Comment:
safe to assume you'll never get `null` ?

Given any thought about enabling `nullable` ? It's very nice to have the compiler tells you about those

Reply:
In this case, yes. The checks happen in the shared code before this is even reached.

But, yes. We must actually enable the nullability checks: https://github.com/xamarin/Essentials/issues/631